CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,24 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * Get Deferred Financial type
+   *
+   * @return array
+   *
+   */
+  public static function getDeferredFinancialType() {
+    $deferredFinancialType = array();
+    $query = "SELECT ce.entity_id, cft.name FROM civicrm_entity_financial_account ce
+INNER JOIN civicrm_option_value cv ON ce.account_relationship = cv.value
```

Comment:
Note that you should be very careful about joining onto the civicrm_option_value table. civicrm_option_value.value is a varchar field. If you join it against an integer field you will get a non-indexed join. I don't think it will matter in this instance since the query is small but in contribution search queries large sites can hit serious problems on some of the joins between financial tables  & option value table. In general it's good to resolve the option value first & then use the value as a constant, rather than joining onto option_value table.

As a rule of thumb you should not join onto the option_value table - even if it's OK in the piece of code you are working on it lends itself to copy & paste.